diffrax/_integrate.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
This addition was necessary to prevent typing errors on [line 80](https://github.com/patrick-kidger/diffrax/pull/357/files#diff-d4d01cfb3cf14986b1415437667bdfecc07730f408e4686a75836cfea459f874R80):
```
progress_bar_state: Optional[PyTree[Array]]
```

Reply:
I think those typing errors are valid: you have e.g. `_TextProgressBarState(progress=0.0)` rather than `_TextProgressBarState(progress=jnp.array(0.0))`.

All adding this here does is to disable some of those runtime type checks, because it can't parse the string as a type! (For this reason I don't use this future import anywhere, as it means that jaxtyping can't do its job.)

Reply:
Ok thank you for these details! I'